clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/loadbalancer/UpsertGoogleLoadBalancerAtomicOperation.groovy

```diff
@@ -139,6 +142,8 @@ class UpsertGoogleLoadBalancerAtomicOperation extends GoogleAtomicOperation<Map>
              || description.healthCheck && !existingTargetPool.healthChecks
              || !description.healthCheck && existingTargetPool.healthChecks)
 
+        needToUpdateSessionAffinity = description.sessionAffinity != GoogleSessionAffinity.valueOf(existingTargetPool.getSessionAffinity());
```

Comment:
Is `description.sessionAffinity` null when when there is no affinity selected? What does the `valueOf()` report from the existing target pool has no session affinity configured?

Reply:
`description.sessionAffinity` is `NONE` when no affinity selected.

`existingTargetPool.getSessionAffinity()` returns `"NONE"`, `"CLIENT_IP"` or `"CLIENT_IP_PROTO"`. `valueof()` returns corresponding enum value. 

We are comparing enum values here.